### PR TITLE
feat: edit blog post endpoint feature

### PR DESCRIPTION
**Pull request by akinwalexander**
### Description
This API endpoint allows super admins to edit existing blog posts. It validates the input data and updates the blog post securely in the database.

### Endpoint
* **Method:** PUT
* **URL:** `/api/v1/blog/edit/:id`

### Request Body
```json
{
  "title": "string", // Required
  "excerpt": "string", // Optional
  "tags": ["string"] // Optional
}
```

## Response
* Success (200 OK):
```JSON
{ 
  "message": "Blog successfully updated",
  "id": "int",
  "title": "string",
  "excerpt": "string",
  "updated_at": "Datatime"
}
```

## Acceptance Criteria
 * The endpoint allows only a super admin to edit an existing blog post.
 * The endpoint should be accessible at /api/v1/blog/edit/{blog_id}
 * The endpoint should accept HTTP PUT requests.
 * The API should validate the request payload to ensure the title and other necessary fields is 
 provided.
 * Upon successful validation of the input data, the blog post should be updated in the database securely.
* Return a 200 OK status code on successful request.
## Request body application/json
``` {
  "title": "string",
  "excerpt": "string",
  "tags": ["string"]
}
```
## Success payload
```{ 
  "message": "Blog successfully updated",
  "id": "int",
  "title": "string",
  "excerpt": "string",
  "updated_at": "Datatime"
}
```
## Error payload
```{
  "message": "Unauthorized request",
  "status_code": 401
}
```

## Requirements
- [ ] Implement server-side logic to handle blog post edit submissions.
- [ ] Validate and sanitize incoming data.
- [ ] Update the blog post securely in the database.
- [ ] A 401 error should be sent if an unauthorized user tries to edit a blog

## Expected Outcome
* The API endpoint allows super admin to edit their blog posts via the provided data, ensuring all updates are securely stored and validated.

## Test
- [x] Write unit tests to ensure the blog post edit endpoint validates input correctly and updates the post securely.
- [x] Test various scenarios for editing blog posts (e.g., missing title, empty content, invalid tags).
- [x] Perform security test to ensure unauthorized user can not update a blog post.
- [x] Test various scenarios:
    - Successful editing of an existing blog post
    - Attempting to edit a non existing blog post
    - Attempting to edit a blog post without authentication
    - Attempting to edit a blog post without proper permissions

## Referrence Issue
[Blog Edit](https://github.com/hngprojects/hng_boilerplate_nestjs/issues/136)

### CODE REVIEW

**Review by rugue — CHANGES_REQUESTED** [review]
Pls resolve your merge conflict, Pull from dev branch before making PR

**Comment by davidolanrewaju** [comment]
And also follow the approved PR template

File: src/modules/blog/controllers/blog.controller.ts [hunk 1]
```diff
@@ -32,4 +34,10 @@ export class BlogController {
   }
 
   //Other blog method endpoint here
+  //edit blog endpoint
+
+  @Put('edit/:id')
```
Comment (davidolanrewaju) [inline src/modules/blog/controllers/blog.controller.ts:39]: Document your routes

File: src/modules/blog/controllers/blog.controller.ts [hunk 2]
```diff
@@ -32,4 +34,10 @@ export class BlogController {
   }
 
   //Other blog method endpoint here
```
Comment (davidolanrewaju) [inline src/modules/blog/controllers/blog.controller.ts:36]: Comments are best removed

File: src/modules/blog/services/blog.service.ts [hunk 3]
```diff
@@ -35,4 +36,30 @@ export class BlogService {
   }
 
   //Other blog service here
```
Comment (davidolanrewaju) [inline src/modules/blog/services/blog.service.ts:38]: Comments are best removed

File: src/modules/blog/controllers/blog.controller.ts (hunk 2 above)
Comment (akinwalexander) [reply src/modules/blog/controllers/blog.controller.ts]: Thank you, they are removed now.

File: src/modules/blog/controllers/blog.controller.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (akinwalexander) [reply src/modules/blog/controllers/blog.controller.ts:36]: Endpoint Documented

**Comment by TheCodeGhinux** [comment]
Fix conflixts sir

**Pull request closed without merging**